File: algorithims/selection_sort.js
//selection_sort, swap and findMin used to create


var swap = function(array, firstIndex, secondIndex) {
    var temp = array[firstIndex];
    array[firstIndex] = array[secondIndex];
    array[secondIndex] = temp;
};

var indexOfMinimum = function(array, startIndex) {

    var minValue = array[startIndex];
    var minIndex = startIndex;

    for(var i = minIndex + 1; i < array.length; i++) {
        if(array[i] < minValue) {
            minIndex = i;
            minValue = array[i];
        }
    } 
    return minIndex;
}; 

// Selection sort vvv  //

var selectionSort = function(array) {
    for (var i = 0; i < array.length; i++){
        var min_ind = indexOfMinimum(array, i);
        if (array[i] > array[min_ind]) {
            swap(array, i, min_ind);
        }
    }
};

var array = [22, 11, 99, 88, 9, 7, 42];
selectionSort(array);

File: algorithims/selection_sort.rb
#selection_sort.rb
require_relative 'find_min.rb'
require_relative 'swap.rb'

def selection_sort(array)
	array.each_with_index do |val, i|
		min_ind = find_index_of_min(array, i)
		if array[min_ind] < array[i]
			swap(array, min_ind, i)
		end
	end 
	array
end

print selection_sort([22, 11, 99, 88, 9, 7, 42])